### ISSUE TITLE
Matrix Testing for Examples

### ISSUE THREAD

**Issue by ryanleecode** [paritytech/capi#942]
For `virtual.eg.ts` we currently test it as is without specifying the `state` flag. How might we approach testing no state flag and with state flag to get full coverage. Same question for other examples with flags. 

This is important if we plan on adding additional flags i.e. `use-stash` and want full test coverage

```ts
const args = parse(Deno.args, { string: ["state"], boolean: ["use-stash"] })
```

**Comment by tjjfvi** [comment]
Perhaps this would be best as an issue on https://github.com/paritytech/egts?

**Comment by harrysolovay** [comment]
Agreed: we can specify variants / alternative inputs in the _frontmatter_/module-level docs.

```ts
/**
 * // ...
 * @alt flags --state <somestatevaluehere>
 */
```

In this case, we specify an alternative run with the flags following `flags`.

We may also want to prefix egts-specific frontmatter fields (including `@skip_test`).

/**
 * // ...
 * @egts:alt flags --state <somestatevaluehere>
 */
```
